packages/utils/src/eventbuilder.ts

```diff
@@ -0,0 +1,132 @@
+import type {
+  Event,
+  EventHint,
+  Exception,
+  Hub,
+  Mechanism,
+  Severity,
+  SeverityLevel,
+  StackFrame,
+  StackParser,
+} from '@sentry/types';
+
+import { isError, isPlainObject } from './is';
+import { addExceptionMechanism, addExceptionTypeValue } from './misc';
+import { normalizeToSize } from './normalize';
+import { extractExceptionKeysForMessage } from './object';
+
+/**
+ * Extracts stack frames from the error.stack string
+ */
+export function parseStackFrames(stackParser: StackParser, error: Error): StackFrame[] {
+  return stackParser(error.stack || '', 1);
+}
+
+/**
+ * Extracts stack frames from the error and builds a Sentry Exception
+ */
+export function exceptionFromError(stackParser: StackParser, error: Error): Exception {
+  const exception: Exception = {
+    type: error.name || error.constructor.name,
+    value: error.message,
+  };
+
+  const frames = parseStackFrames(stackParser, error);
+  if (frames.length) {
+    exception.stacktrace = { frames };
+  }
+
+  return exception;
+}
+
+/**
+ * Builds and Event from a Exception
+ * @hidden
+ */
+export function eventFromUnknownInput(
+  getCurrentHub: () => Hub,
+  stackParser: StackParser,
+  exception: unknown,
+  hint?: EventHint,
+): Event {
+  let ex: unknown = exception;
+  const providedMechanism: Mechanism | undefined =
+    hint && hint.data && (hint.data as { mechanism: Mechanism }).mechanism;
+  const mechanism: Mechanism = providedMechanism || {
+    handled: true,
+    type: 'generic',
+  };
+
+  if (!isError(exception)) {
+    if (isPlainObject(exception)) {
+      // This will allow us to group events based on top-level keys
+      // which is much better than creating new group when any key/value change
+      const message = `Non-Error exception captured with keys: ${extractExceptionKeysForMessage(exception)}`;
```

Comment:
l: I wonder if we want to align this with the changes we did for browser-based non-error-object capturing: https://github.com/getsentry/sentry-javascript/pull/8374

But maybe we do that in a separate follow-up PR!

Reply:
There has been some discussion around consolidating the node/browser eventbuilders too...

Reply:
let's do this in a follow up PR after we merge in node sdk changes re: server runtime!